src/MongoDB/Manager.c

```diff
@@ -669,22 +671,53 @@ static PHP_METHOD(Manager, startSession)
 		return;
 	}
 
-	if (options && php_array_exists(options, "causalConsistency")) {
+	if (options && php_array_existsc(options, "causalConsistency")) {
 		cs_opts = mongoc_session_opts_new();
 		mongoc_session_opts_set_causal_consistency(cs_opts, php_array_fetchc_bool(options, "causalConsistency"));
 	}
 
-	cs = mongoc_client_start_session(intern->client, cs_opts, &error);
+	if (options && php_array_existsc(options, "defaultTransactionOptions")) {
+		zval* txn_options = php_array_fetchc(options, "defaultTransactionOptions");
 
-	if (cs_opts) {
-		mongoc_session_opts_destroy(cs_opts);
+		/* Thrown exception and return if the defaultTransactionOptions is not an array */
+		if (Z_TYPE_P(txn_options) != IS_ARRAY) {
+			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected \"defaultTransactionOptions\" option to be an array");
+			goto cleanup;
+		}
+
+		/* Parse transaction options */
+		txn_opts = php_mongodb_session_parse_transaction_options(txn_options TSRMLS_CC);
+
+		/* If an exception is thrown while parsing, destroy options if allocated, and return */
+		if (EG(exception)) {
+			goto cleanup;
+		}
+
+		/* If the options are non-empty, add them to the client session opts struct */
+		if (txn_opts) {
+			if (!cs_opts) {
+				cs_opts = mongoc_session_opts_new();
+			}
+
+			mongoc_session_opts_set_default_transaction_opts(cs_opts, txn_opts);
```

Comment:
Can we call `mongoc_transaction_opts_destroy()` after this? Ideally, this should be the only point where `txn_opts` is non-NULL and in need of freeing (provided the memory leak in https://github.com/mongodb/mongo-php-driver/pull/850/files#r196198256 is fixed).

Reply:
Yes, done. And updated relevant comments.